### PR TITLE
[14.0][mig]-account_analytic_journal_lock_to_date

### PR DESCRIPTION
**Pull request by JuanyDForgeflow**

### CODE REVIEW

File: account_analytic_journal_lock_to_date/models/account_analytic_line.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            if (
+                not line.journal_id.restrict_lock_dates
+                or line.account_id.surpass_lock_dates
+            ):
+                continue
+            lock_date = line.journal_id.journal_lock_to_date
+            if not lock_date:
+                lock_date = datetime.now().date() + timedelta(days=1)
+            else:
+                lock_date = datetime.strptime(str(lock_date), DT).date()
```
Comment (JasminSForgeFlow) [inline account_analytic_journal_lock_to_date/models/account_analytic_line.py]: Here I think you can use lock_date directly, no need to convert

**Review by JasminSForgeFlow — CHANGES_REQUESTED** [review]

File: account_analytic_journal_lock_to_date/models/account_analytic_line.py (hunk 1 above)
Comment (JuanyDForgeflow) [reply account_analytic_journal_lock_to_date/models/account_analytic_line.py]: I was getting an error saying it expected string instead of datetime object

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
It needs fixing I can create analytic lines after the lock dates of the analytic journal, that should not be the case

File: account_analytic_journal_lock_to_date/models/account_analytic_line.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply account_analytic_journal_lock_to_date/models/account_analytic_line.py]: This module is very similar and the conversion is not needed: https://github.com/OCA/account-financial-tools/blob/14.0/account_lock_to_date/models/account_move.py#L29 could you try this?

**Comment by JuanyDForgeflow** [comment]
@AaronHForgeFlow it's fixed now please try again 
image

File: account_analytic_journal_lock_to_date/models/account_analytic_line.py (hunk 1 above)
Comment (JuanyDForgeflow) [reply account_analytic_journal_lock_to_date/models/account_analytic_line.py]: @JasminSForgeFlow @AaronHForgeFlow I changed it, now it's working

**Review by AaronHForgeFlow — APPROVED** [review]

**Review by JasminSForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
This branch has conflicts because of the changes in the analytic_plan module, can you fix that so I can merge?

**Comment by AaronHForgeFlow** [comment]
It seems there is some issue with github. Author of the commit is @JuanyDForgeflow  not me. It shows her in the commits tab so I think the history is fine.

**Pull request merged**